torch/nn/init.py

```diff
@@ -190,6 +190,8 @@ def constant_(tensor: Tensor, val: float) -> Tensor:
         >>> w = torch.empty(3, 5)
         >>> nn.init.constant_(w, 0.3)
     """
+    if has_torch_function_variadic(tensor):
```

Comment:
Isn't this already catching the `.fill_()` function? Why do we want this here?

Reply:
I think this calls tensor.fill_() under the hood and not torch.fill_(). I tried a simple test and it doesn't seem to work:

```
import torch

class MyTensor():

    def __torch_function__(self, func, types, args=(), kwargs=None):
        print(f'called: {func}')

t = MyTensor()
torch.nn.init.constant_(t, 0)
```

The error is:
```
'MyTensor' object has no attribute 'fill_'
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
 in <module>
      7 
      8 t = MyTensor()
----> 9 torch.nn.init.constant_(t, 0)
torch/nn/init.py in constant_(tensor, val)
    191         >>> nn.init.constant_(w, 0.3)
    192     """
--> 193     return _no_grad_fill_(tensor, val)
    194 
    195 
torch/nn/init.py in _no_grad_fill_(tensor, val)
     60 def _no_grad_fill_(tensor, val):
     61     with torch.no_grad():
---> 62         return tensor.fill_(val)
     63 
     64 
AttributeError: 'MyTensor' object has no attribute 'fill_'
```

Reply:
Right if you're not using a Tensor subclass, you don;t get any method "for free" you have to add them all. That is expected.